Add blog post for deprecated backports and typing aliases (#2139)
home-assistant/developers.home-assistant: blog/2024-04-08-deprecated-backports-and-typing-aliases.md

---
author: Marc Mueller
authorURL: https://github.com/cdce8p
title: "Deprecate old backports and typing alias"
---

> In the past we've backported features from upstream CPython to be able to use them early and improve the experience for both users and developers. Now that Home Assistant only supports Python 3.12, these can be used from Python directly. These backports are now deprecated and will be removed in the future.

Reviewer: ```suggestion
In the past, we've backported features from upstream CPython to use them early and improve user and developers' experience. Home Assistant only supports Python 3.12, so these can be used directly from Python. These backports are now deprecated and will be removed in the future.
```

| Deprecated | Replacement | Python version |
| ---------- | ----------- | -------------- |
| `homeassistant.backports.enum.StrEnum` | `enum.StrEnum` | >= 3.11 |
| `homeassistant.backports.functools.cached_property` | `functools.cached_property` | >= 3.8, >= 3.12 (performance improvement) |

> In addition, some typing alias are also deprecated now.

Reviewer: ```suggestion
In addition, some typing aliases are also deprecated now.
```
